src/Cli/dotnet/commands/dotnet-workload/install/NetSdkManagedInstaller.cs

```diff
@@ -145,6 +145,9 @@ public void InstallWorkloadPack(PackInfo packInfo, SdkFeatureBand sdkFeatureBand
             }
             finally
             {
+                GC.Collect();
+                GC.WaitForPendingFinalizers();
```

Comment:
Note that in general, this pattern isn't sufficient, if you have chains of finalizable objects, you might need to do this `n` times where `n` is the length of the longest possible chain.

Is there no disposable based system that can be used here instead?

Reply:
yeah, this is not ideal. We're seeing lots of issues with quick, successive file operations where handles are not being released, e.g. we extract a file, read it, close it, delete it, and then delete fails, even after explicitly calling close() before deleting it.